main.cpp

```diff
@@ -51,22 +52,58 @@ const char *host_name = "echo.mbedcloudtesting.com";
 // Echo server port (same for TCP and UDP)
 const int port = 7;
 
-Mutex PrintMutex;
+static rtos::Mutex trace_mutex;
+
+#if MBED_CONF_MBED_TRACE_ENABLE
+static void trace_wait()
+{
+    trace_mutex.lock();
+}
+
+static void trace_release()
+{
+    trace_mutex.unlock();
+}
+
+
+static uint32_t cellular_starttime = us_ticker_read() / 1000L;
```

Comment:
This timer won't run if power saving is in use. I suggest `Kernel::get_ms_count()` if you're looking for something bearing a firm relation to real time. And then you shouldn't need the start time either, as it starts at 0 on boot.

Reply:
good to know, changed to Kernel::get_ms_count()